src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ARRAY_LENGTH_MULTIPLIER = 1.5;
+    private int size;
+    private Object[] majorArray;
+    private T[] oldArray;
+    private T removedElement;
+
+    public ArrayList() {
+        majorArray = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == majorArray.length) {
+            increasingMajorArray();
+        }
+        majorArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("The index is outside the array");
+        } else {
+            if (size + 1 >= majorArray.length) {
+                increasingMajorArray();
+            }
+            System.arraycopy(majorArray, index, majorArray, index + 1, size - index);
+            majorArray[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int k = 0; k < list.size(); k++) {
+            if (size + list.size() == majorArray.length) {
+                increasingMajorArray();
+            }
+            majorArray[k + size] = list.get(k);
+        }
+        size = size + list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("The index is outside the array");
+        } else {
+            return (T) majorArray[index];
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("The index is outside the array");
```

Comment:
<img width="1017" alt="image" src="https://github.com/mate-academy/jv-array-list/assets/81781479/fd902e2d-181f-4d2b-a437-c76519ae6b30">

move the check to another method